packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -145,33 +152,88 @@ export class ApolloServer extends ApolloServerBase {
-          result && {
-            ...result,
-            headers: {
-              ...result.headers,
-              ...corsHeaders,
+      const makeCallbackFilter = (
+        doneFn: any,
+      ): lambda.APIGatewayProxyCallback => {
+        return (error, result) => {
+          doneFn();
+          callback(
+            error,
+            result && {
+              ...result,
+              headers: {
+                ...result.headers,
+                ...corsHeaders,
+              },
             },
-          },
-        );
+          );
+        };
       };
 
-      graphqlLambda(async () => {
-        // In a world where this `createHandler` was async, we might avoid this
-        // but since we don't want to introduce a breaking change to this API
-        // (by switching it to `async`), we'll leverage the
-        // `GraphQLServerOptions`, which are dynamically built on each request,
-        // to `await` the `promiseWillStart` which we kicked off at the top of
-        // this method to ensure that it runs to completion (which is part of
-        // its contract) prior to processing the request.
-        await promiseWillStart;
-        return this.createGraphQLServerOptions(event, context);
-      })(event, context, callbackFilter);
+      const response = new Stream.Writable();
+
+      fileUploadProcess(event, response, this.uploadsConfig || {})
+        .then((body: any) => {
+          event.body = body;
+          graphqlLambda(async () => {
+            // In a world where this `createHandler` was async, we might avoid this
+            // but since we don't want to introduce a breaking change to this API
+            // (by switching it to `async`), we'll leverage the
+            // `GraphQLServerOptions`, which are dynamically built on each request,
+            // to `await` the `promiseWillStart` which we kicked off at the top of
+            // this method to ensure that it runs to completion (which is part of
+            // its contract) prior to processing the request.
+            await promiseWillStart;
+            return this.createGraphQLServerOptions(event, context);
+          })(
+            event,
+            context,
+            makeCallbackFilter(() => {
+              response.end();
+            }),
+          );
+        })
+        .catch(error => {
+          throw formatApolloErrors([error], {
+            formatter: this.requestOptions.formatError,
+            debug: this.requestOptions.debug,
+          });
+        });
     };
   }
 }
+
+const fileUploadProcess = (
+  event: any,
+  response: any,
+  uploadsConfig: FileUploadOptions,
+) => {
+  return new Promise((resolve, reject) => {
+    const contentType =
+      event.headers['content-type'] || event.headers['Content-Type'];
+
+    if (
+      typeof processFileUploads === 'function' &&
+      contentType &&
+      contentType.startsWith('multipart/form-data')
+    ) {
+      const request = new stream.Readable() as any;
```

Comment:
I have an error here. got it work by changing to new Stream.Readable().  
typo ?

Reply:
@charleswong28 Would like to confirm with you, any comments on this ?